### DISCUSSION TITLE
Tracking events by referrer / UTM

### DISCUSSION THREAD

**Discussion by funkhaus-wout** [umami-software/umami#2957] [General]
Hello,

Is there any way to track certain events by referrer or UTM?
Our clients generally wants to know how many events they received from a given Google ad / Meta ad / Social media post / …

It would be Great if the events section had a filter to do just that!
Any ideas?

**Comment by FHU-yezi** [comment]
Maybe you can just use the URL params filter?

**Comment by funkhaus-wout** [comment]
Thanks for the tip FHU-yezi, but I'm still not sure how to achieve my goal.
As I said, it would be great if we can select an event, and see where these visitors came from (Google ad / Meta ad / Social media post / organic / …)

I don't think this is possible yet right?

**Comment by gigor** [comment]
I wrote a little wrapper, it saves utm tags and other params that I am interested in to LocalStorage and attaches them to every event.  [Here is a gist](https://gist.github.com/gigor/ff904e892d97676a2b9e0cad50cf02a5)

This way, even if the event is completed on a second visit, I will know where it came from originaly.

**Reply by funkhaus-wout** [reply]
Great stuff! Thanks for your help so far :)
